src/checkpoint/container/array_serialize.h

```diff
@@ -54,6 +54,7 @@ namespace checkpoint {
 
 template <typename Serializer, typename T, size_t N>
 void serialize(Serializer& s, std::array<T, N>& array) {
+  s.countBytes(array);
   dispatch::serializeArray(s, &array[0], array.size());
 }
```

Comment:
I think this is double-counting, since the implementation is required to contain the `T[N]` as a member, not point to it. So, I think the `countBytes` can just be omitted

Reply:
removed the redundant `countBytes` call